### ISSUE TITLE
Bug in negation predicate

### ISSUE THREAD

**Issue by mingodad** [yhirose/cpp-peglib#210]
While trying this grammar:
```
STRING_LONG <-
	 '[' '='* '['  ( ! (']' '='* ']') . )* ']' '='* ']'
```
With this input:
```
[==[One string]]==]
```
I'm getting this error, while I was expecting none, and the AST is shown correctly:
```
1:16 syntax error, unexpected ']', expecting '='.
```
See discussion here: https://github.com/ChrisHixon/chpeg/issues/17#issuecomment-1146876233

**Comment by ChrisHixon** [comment]
I believe this is a misunderstanding in how PEG works. The `'='` being optional with `*` inside the negation predicate makes it work like `!(']' ']')` or simply `!']]'`. The inner string is matching to the g in string, then one `]` matches, then zero `*`, then one `]`. It's left with `==]` in the input. cpp-peglib doesn't seem to indict there are excess characters left in the input.

The error in chpeg from this:
```
Extraneous input: parse consumed 16 bytes out of 19
  Expected: "=" in STRING_LONG at offset 15
```

**Comment by yhirose** [comment]
cpp-peglib works exactly as @ChrisHixon mentioned.

Here is the result (I added EOF `!.` at the end for better error position) on cpp-peglib playground:
<img width="1070" alt="image" src="https://user-images.githubusercontent.com/357397/172277492-517f449a-35f2-4a06-8b1b-d717c4dacccc.png">

I think this is the natural PEG behavior since I faithfully implemented according to the Bryan Ford's paper, and I didn't do anything special. In order to accomplish what you probably want to do, I added another 'not predicate' operator.

<img width="911" alt="image" src="https://user-images.githubusercontent.com/357397/172280131-7b7d80f6-bc9c-490f-95d1-6555aa601679.png">

I am not sure if it covers all the situation that you are thinking of. But at least it fixes the error in your example. Hope it helps.
